lib/net.js

```diff
@@ -583,26 +583,32 @@ Socket.prototype._destroy = function(exception, cb) {
     clearTimeout(s[kTimeout]);
   }
 
+  cb();
+
+  const onClosed = () => {
+    debug('emit close');
+    if (exception) {
+      this.emit('error', exception)
```

Comment:
If I'm not wrong this will emit an error every time `socket.destroy()` is called with an error.

Reply:
That will happen through `cb(exception)` as well?

Reply:
yes, you are right, this is quite complicated...